server/models/campaign.js

```diff
@@ -0,0 +1,59 @@
+const mongoose = require("mongoose");
+const Schema = mongoose.Schema;
+
+const CampaignSchema = new Schema({
+  name: { type: String, required: true },
+  created: { type: Date, default: Date.now, required: true },
+  ownedBy: { type: Schema.Types.ObjectId, ref: "User", index: true },
+  prospects: [{ type: Schema.Types.ObjectId, ref: "Prospect", index: true }],
```

Comment:
We will need statuses here as well - so making it another schema, that is used here and inside the steps would be great

I actually liked how it was a separate model though referenced by id, and how that model had a list itself that was embedded. The reason is because we don't want each campaign to be massive, especially since we will be getting a lot of them for each user

Reply:
ok, makes sense. I've split it back out